### ISSUE TITLE
Compile-time warnings

### ISSUE THREAD

**Issue by gergelypolonkai** [clemera/objed#22]
During byte-compilation, i get these warnings:

```
objed.el:168:1:Warning: defcustom for ‘objed-init-p-function’ fails to specify
    type
objed.el:168:1:Warning: defcustom for ‘objed-init-p-function’ fails to specify
    type
objed.el:280:1:Warning: defcustom for ‘objed-modeline-hint-p’ fails to specify
    type
objed.el:280:1:Warning: defcustom for ‘objed-modeline-hint-p’ fails to specify
    type

In objed--ob-apply:
objed.el:2530:6:Warning: objed-exit-op called with 3 arguments, but accepts
    only 1-2
```

The last one seems a bit more serious than the others, although i don’t really understand why it is coming, as the function is defined as

```
(defun objed-exit-op (op &optional text range) …)
```

The first four is easily mitigated by adding `:type 'function` to the former and `:type 'boolean` to the latter.

**Comment by clemera** [comment]
HI, thanks for the report. I don't get those, which Emacs version are you using?

**Comment by gergelypolonkai** [comment]
That’s exactly what’s happening; i don’t know if it’s actually a bug, though.

To make the warning less confusing, you can move the `defun` before the very first invocation. In this case the warning will be less scary:

```
Warning: function objed-exit-op used to take 2 arguments, now takes 3
```

**Comment by clemera** [comment]
Ok, I guess that means the compiler just does not know the function definition has changed when code that uses that function gets compiled. But avoiding this would mean to always move function definitions ahead of their invocation. I think it's not a bug and those warning can just be ignored after an update?

**Comment by gergelypolonkai** [comment]
No, it’s definitely not a bug an yes, it’s safe to ignore this during an upgrade. Still, the `now takes 3` version is less confusing, asd people as suspicious as me would more likely ignore that than the original one :)

**Comment by clemera** [comment]
Ok, fixed by e9bfd9239b560a7065d303ea5cc2b964f1ebf98d!
